src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using global::Iot.Units;
+
+namespace Iot.Device.Ad7193
+{
+    /// <summary>
+    /// Represents the Analog Devices AD7193, the 4-channel, 4.8 kHz, ultralow noise, 24-bit sigma-delta ADC with PGA
+    /// </summary>
+    public class Ad7193 : IDisposable
+    {
+        #region Metadata for IDevice
+
+        /// <summary>
+        /// Name of the manufacturer of the device
+        /// </summary>
+        public const string Manufacturer = "Analog Devices";
+
+        /// <summary>
+        /// Name of the product
+        /// </summary>
+        public const string Product = "AD7193";
+
+        /// <summary>
+        /// Category of the device
+        /// </summary>
+        public const string ProductCategory = "ADC";
+
+        /// <summary>
+        /// Description of the device
+        /// </summary>
+        public const string ProductDescription = "4-Channel, 4.8 kHz, Ultralow Noise, 24-Bit Sigma-Delta ADC with PGA";
+
+        /// <summary>
+        /// The URI of the datasheet of the device
+        /// </summary>
+        public const string DataSheetURI = "https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf";
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
+
+        /// <summary>
+        /// The maximum frequency that can be used on the SPI bus
+        /// </summary>
+        public const int MaximumSpiFrequency = 10000000;
+        #endregion
+
+        #region metadata for IAdcDevice
+
+        /// <summary>
+        /// Number of ADCs on the device
+        /// </summary>
+        public const int ADCCount = 1;
+
+        /// <summary>
+        /// The maximum bitrate of the ADC
+        /// </summary>
+        public const int ADCBitrate = 24;
+
+        /// <summary>
+        /// The maximum sampling rate of the ADC
+        /// </summary>
+        public const int ADCSamplerate = 4800;
+
+        /// <summary>
+        /// The number of channels the ADC has
+        /// </summary>
+        public const int ADCInputChannelCount = 8;
+        #endregion
+
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        // protected ConcurrentQueue<AdcValue> adcValues = new ConcurrentQueue<AdcValue>();
+        protected BlockingCollection<AdcValue> adcValues = new BlockingCollection<AdcValue>();
+
+        /// <summary>
+        /// The event that is fired every time a new value is received from the ADC
+        /// </summary>
+        public event EventHandler<AdcValueReceivedEventArgs> AdcValueReceived;
```

Comment:
NIT: should the name of this event be more something like `OnValueChange` or something like that?

Reply:
I don't think so, but I'm not a native speaker, you are probably a better judge of this :)

When the event fires, the old value doesn't change, but a new value got converted and was added to the list of values in the buffer. 
It's more like when in network communications we receive a new packet that we potentially need to process it.

Reply:
I see, my comment wasn't really about using proper english (I'm also not a native speaker 😄), but it was instead more about how EventHandlers traditionally look in C# APIs. They usually have the pattern of "OnSomethingHappened" so that it is clear that this is an eventhandler and that people that want to use it should just append delegates to it. Given your explanation, probably the right name for it should be `OnValueReceived`?

Reply:
Oh, I see, sure! :) I renamed it.